sled-agent/src/storage_manager.rs

```diff
@@ -129,28 +129,23 @@ pub enum Error {
 
 /// A ZFS storage pool.
 struct Pool {
-    id: Uuid,
+    name: ZpoolName,
     info: ZpoolInfo,
     // ZFS filesytem UUID -> Zone.
     zones: HashMap<Uuid, RunningZone>,
-
-    // The devfs path to the disk where this zpool exists.
-    disk_path: Option<PathBuf>,
+    parent: DiskIdentity,
 }
 
 impl Pool {
     /// Queries for an existing Zpool by name.
     ///
     /// Returns Ok if the pool exists.
-    fn new(
-        name: &ZpoolName,
-        disk_path: Option<PathBuf>,
-    ) -> Result<Pool, Error> {
+    fn new(name: ZpoolName, parent: DiskIdentity) -> Result<Pool, Error> {
         let info = Zpool::get_info(&name.to_string())?;
 
         // NOTE: This relies on the name being a UUID exactly.
         // We could be more flexible...
-        Ok(Pool { id: name.id(), info, zones: HashMap::new(), disk_path })
+        Ok(Pool { name, info, zones: HashMap::new(), parent })
```

Comment:
The comment above this line seems not quite true?

Reply:
Comment removed